rego/Drive.rego

```diff
@@ -134,12 +136,34 @@ if {
 #
 # Baseline GWS.DRIVEDOCS.1.4v0.1
 #--
+NoSuchEvent1_4(TopLevelOU) := true if {
+    SettingName := "SHARING_INVITES_TO_NON_GOOGLE_ACCOUNTS"
+    Events_A := utils.FilterEvents(LogEvents, SettingName, TopLevelOU)
+    count(Events_A) == 0
+}
+
+NoSuchEvent1_4(TopLevelOU) := true if {
+    SettingName := "SHARING_OUTSIDE_DOMAIN"
+    Events_B := utils.FilterEvents(LogEvents, SettingName, TopLevelOU)
+    count(Events_B) == 0
+}
+
+default NoSuchEvent1_4(_) := false
+
 NonCompliantOUs1_4 contains OU if {
     some OU in utils.OUsWithEvents
-    Events := utils.FilterEvents(LogEvents, "SHARING_INVITES_TO_NON_GOOGLE_ACCOUNTS", OU)
-    count(Events) > 0
-    LastEvent := utils.GetLastEvent(Events)
-    contains("NOT_ALLOWED INHERIT_FROM_PARENT", LastEvent.NewValue) == false
+    Events_A := utils.FilterEvents(LogEvents, "SHARING_INVITES_TO_NON_GOOGLE_ACCOUNTS", OU)
+    count(Events_A) > 0
+    LastEvent_A := utils.GetLastEvent(Events_A)
+
+    Events_B := utils.FilterEvents(LogEvents, "SHARING_OUTSIDE_DOMAIN", OU)
+    count(Events_B) > 0
+    LastEvent_B := utils.GetLastEvent(Events_B)
+
+    AcceptableValues_A := {"NOT_ALLOWED", "INHERIT_FROM_PARENT"}
+    not LastEvent_A.NewValue in AcceptableValues_A
+    AcceptableValues_B := {"SHARING_ALLOWED_WITH_WARNING", "SHARING_NOT_ALLOWED", "INHERIT_FROM_PARENT"}
```

Comment:
`SHARING_ALLOWED_WITH_WARNING` should not be in the list of AcceptableValues_B. Otherwise, the following configuration results in a pass for 1.4:
![image](https://github.com/cisagov/ScubaGoggles/assets/106177711/881b815c-f0ae-4fb1-90d4-2bf5b0fbd95f)

```suggestion
    AcceptableValues_B := {"SHARING_NOT_ALLOWED", "INHERIT_FROM_PARENT"}
```